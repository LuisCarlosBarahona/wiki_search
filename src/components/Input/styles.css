.input-field {
    background: #fff;
    display: flex;
    box-shadow: none;
    padding: 15px 30px;
    border: 1px solid grey;
    border-radius: 24px;
    font-size: 24px;
    height: 44px;
    padding: 0 30px;
    width: 400px;
}

.input-field:focus {
    box-shadow: 0 1px 6px 0 rgba(32,36,33,0.28);
    border-color: rgba(223, 225, 229, 0);
    outline: none;
}