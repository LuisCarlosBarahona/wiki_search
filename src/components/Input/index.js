import React from 'react';

import './styles.css';
const Input = ({placeholder, ...rest}) => 
    <input className="input-field" placeholder={placeholder} {...rest}/>

Input.defaultProps = {
    placeholder: 'Input search query'
};

export default Input;