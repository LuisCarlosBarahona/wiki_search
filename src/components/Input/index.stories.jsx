import React from 'react';

import Input from '.';

export default {
  title: 'Components/Input',
  component: Input,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
};

const Template = (args) => <Input {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  placeholder: 'Search for data'
};
