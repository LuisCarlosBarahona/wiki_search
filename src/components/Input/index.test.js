import Input from '.';
import {render} from '@testing-library/react'

describe('Input Component', () => {
    let sut;
    let props;

    it('should match snapshot', () => {
        sut = render(<Input {...props}/>);
        expect(sut).toMatchSnapshot();
    });
});