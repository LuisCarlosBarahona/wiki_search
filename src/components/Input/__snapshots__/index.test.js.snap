// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component with props should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Input Component without props should match snapshot 1`] = `ShallowWrapper {}`;
